test/battle/ability/teraform_zero.c

```diff
@@ -0,0 +1,102 @@
+{
+    GIVEN {
+        PLAYER(SPECIES_TERAPAGOS_TERASTAL);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_KYOGRE) {Ability(ABILITY_DRIZZLE); }
+        OPPONENT(SPECIES_TAPU_KOKO) {Ability(ABILITY_ELECTRIC_SURGE); }
+    } WHEN {
+        TURN { MOVE(playerLeft, MOVE_CELEBRATE, gimmick: GIMMICK_TERA); }
+    } SCENE {
+        ABILITY_POPUP(playerLeft, ABILITY_TERAFORM_ZERO);
+        MESSAGE("The rain stopped.");
+        MESSAGE("The electricity disappeared from the battlefield.");
+    }
+}
+
+DOUBLE_BATTLE_TEST("Teraform Zero can be supressed")
+{
+    GIVEN {
+        PLAYER(SPECIES_TERAPAGOS_TERASTAL);
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_WEEZING) {Ability(ABILITY_NEUTRALIZING_GAS); }
+        OPPONENT(SPECIES_KYOGRE) {Ability(ABILITY_DRIZZLE); }
+        OPPONENT(SPECIES_TAPU_KOKO) {Ability(ABILITY_ELECTRIC_SURGE); }
+    } WHEN {
+        TURN { SWITCH(playerRight, 2); MOVE(playerLeft, MOVE_CELEBRATE, gimmick: GIMMICK_TERA); }
+    } SCENE {
+        ABILITY_POPUP(playerRight, ABILITY_NEUTRALIZING_GAS);
+        NONE_OF {
+            MESSAGE("The rain stopped.");
+            MESSAGE("The electricity disappeared from the battlefield.");
+        }
+    }
+}
+
+SINGLE_BATTLE_TEST("Teraform Zero can be replaced")
+{
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_WORRY_SEED].effect == EFFECT_WORRY_SEED);
+        ASSUME(gMovesInfo[MOVE_REST].effect == EFFECT_REST);
+        PLAYER(SPECIES_TERAPAGOS);
+        OPPONENT(SPECIES_WHIMSICOTT) { Ability(ABILITY_PRANKSTER); }
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_WORRY_SEED); MOVE(player, MOVE_REST, gimmick: GIMMICK_TERA); }
+    } SCENE {
+        MESSAGE("Foe Whimsicott used Worry Seed!");
+        MESSAGE("Terapagos acquired Insomnia!");
+        MESSAGE("Terapagos used Rest!");
+        ABILITY_POPUP(player, ABILITY_INSOMNIA);
+        MESSAGE("Terapagos stayed awake using its Insomnia!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Teraform Zero cannot be swapped")
+{
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_SKILL_SWAP].effect == EFFECT_SKILL_SWAP);
+        PLAYER(SPECIES_TERAPAGOS);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_CELEBRATE, gimmick: GIMMICK_TERA); MOVE(opponent, MOVE_SKILL_SWAP); }
+    } SCENE {
+        MESSAGE("Foe Wobbuffet used Skill Swap!");
+        MESSAGE("But it failed!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Teraform Zero cannot be copied")
+{
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_ROLE_PLAY].effect == EFFECT_ROLE_PLAY);
+        PLAYER(SPECIES_TERAPAGOS);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_CELEBRATE, gimmick: GIMMICK_TERA); MOVE(opponent, MOVE_ROLE_PLAY); }
+    } SCENE {
+        MESSAGE("Foe Wobbuffet used Role Play!");
+        MESSAGE("But it failed!");
+    }
+}
+
+DOUBLE_BATTLE_TEST("Teraform Zero shouldn't cause Neutralizing Gas to show it's popup when trying to activate")
```

Comment:
@hedara90 Can you check if the following change fixes it?
```
-    if (type < 32) // avoid OOB access
+    if (type < 32 && gBattleMons[battler].species != SPECIES_TERAPAGOS_STELLAR) // avoid OOB access
```
This change is in file `src/battle_terastal.c` function `ExpendTypeStellarBoost`

Reply:
![20240722_19h08m54s_grim](https://github.com/user-attachments/assets/9fd2f455-e5c6-4522-8fde-f4de0ac449f1)
It seems to fix it.

Reply:
Can you add the fix to this pr?